src/library/scala/collection/immutable/IntMap.scala

```diff
@@ -409,8 +409,8 @@ sealed abstract class IntMap[+T] extends AbstractMap[Int, T]
     * @param f       The function used to resolve conflicts between two mappings.
     * @return        Union of `this` and `that`, with identical key conflicts resolved using the function `f`.
     */
-  def unionWith[S >: T](that: IntMap[S], f: (Int, S, S) => S): IntMap[S] = (this, that) match{
-    case (IntMap.Bin(p1, m1, l1, r1), that@(IntMap.Bin(p2, m2, l2, r2))) =>
+  def unionWith[S >: T](that: IntMap[S], f: (Int, S, S) => S): IntMap[S] = (this, that) match {
+    case (IntMap.Bin(p1, m1, l1, r1), IntMap.Bin(p2, m2, l2, r2)) =>
```

Comment:
presumably no longer would warn, but is one form preferable? it compiles. Also I presume `that` was a `Bin` before, but the syntax is noisy.